test/integration/idle_timeout_integration_test.cc

```diff
@@ -178,8 +193,94 @@ TEST_P(IdleTimeoutIntegrationTest, PerStreamIdleTimeoutAfterBidiData) {
 
 // Successful request/response when per-stream idle timeout is configured.
 TEST_P(IdleTimeoutIntegrationTest, PerStreamIdleTimeoutRequestAndResponse) {
+  enable_per_stream_idle_timeout_ = true;
+  testRouterRequestAndResponseWithBody(1024, 1024, false, nullptr);
+}
+
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutConfiguredRequestResponse) {
+  enable_request_timeout_ = true;
```

Comment:
Should you also test enabling both?  (Up to you; just curious.)

Reply:
ehhh, would actually be testing all three. Started writing one but tbh the separate timeout behaviours should not affect one another.

Reply:
Sure, but you test thing to make sure your assumptions are correct :-}.  (Still up to you, just reacting to the "should not affect" :-}.)